x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetStackTracesAction.java

```diff
@@ -881,10 +881,10 @@ private DetailsHandler(
             this.expectedSlices = new AtomicInteger(expectedExecutableSlices + expectedStackFrameSlices);
         }
 
-        public void onStackFramesResponse(MultiGetResponse multiGetItemResponses) {
+        void onStackFramesResponse(MultiGetResponse multiGetItemResponses) {
             for (MultiGetItemResponse frame : multiGetItemResponses) {
                 if (frame.isFailed()) {
-                    submitListener.onFailure(frame.getFailure().getFailure());
+                    finishWithFirstFailure(frame.getFailure().getFailure());
```

Comment:
Why do we finish with the first instead of collecting all the failed frames?

Reply:
👍 thx, I didn't simply didn't consider changing the behavior, but fixed and adjusted to properly report all suppressed failures